src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -977,13 +954,34 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			connection.extensionTimer.stop();
 
 			connection.connectHandler(true);
-			this.addTelemetryForConnection(connection);
+			this._telemetryService.createActionEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.DatabaseConnected)
+				.withAdditionalProperties({
+					connectionType: connection.serverInfo ? (connection.serverInfo.isCloud ? 'Azure' : 'Standalone') : '',
+					provider: connection.connectionProfile.providerName,
+					serverVersion: connection.serverInfo ? connection.serverInfo.serverVersion : '',
+					serverEdition: connection.serverInfo ? connection.serverInfo.serverEdition : '',
+					serverEngineEdition: connection.serverInfo ? connection.serverInfo.engineEditionId : '',
+					isBigDataCluster: connection.serverInfo?.options?.isBigDataCluster ?? false,
+				}).withAdditionalMeasurements({
+					extensionConnectionTimeMs: connection.extensionTimer.elapsed() - connection.serviceTimer.elapsed(),
+					serviceConnectionTimeMs: connection.serviceTimer.elapsed()
+				})
+				.send();
 
 			if (this._connectionStatusManager.isDefaultTypeUri(info.ownerUri)) {
 				this._connectionGlobalStatus.setStatusToConnected(info.connectionSummary);
 			}
 		} else {
 			connection.connectHandler(false, info.errorMessage, info.errorNumber, info.messages);
+			this._telemetryService.createErrorEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.DatabaseConnectionError, info.errorNumber.toString())
+				.withAdditionalProperties({
+					provider: connection.connectionProfile.providerName,
+				})
```

Comment:
It would be nice to add the errorNumber or some information from the error so it's more insightful than just number of failures.

Reply:
`info.errorNumber.toString()` is in the createErrorEvent parameters

Eventually I might come back to this and try to pull out more information from the error message (look for patterns of the most common errors) since the error number isn't always going to be the most useful here - but I want to get this in ASAP since the primary purpose at this point is just to capture how many failures we're getting in general so that we can easily tell if there's widespread connection issues with a new release like has happened a few times recently.